### PR TITLE
DCA-1016: Confluence prepare data script update

### PR DESCRIPTION
**Pull request by opopovss**

### CODE REVIEW

File: app/util/data_preparation/confluence_prepare_data.py [hunk 1]
```diff
@@ -3,12 +3,15 @@
 
 import urllib3
 
+from multiprocessing import cpu_count
```
Comment (ometelytsia) [inline app/util/data_preparation/confluence_prepare_data.py:6]: Not used import.

File: app/util/data_preparation/confluence_prepare_data.py [hunk 2]
```diff
@@ -3,12 +3,15 @@
```
Comment (ometelytsia) [inline app/util/data_preparation/confluence_prepare_data.py:3]: There are a bunch of missing imports related to the `print_timing` decorator.

File: app/util/data_preparation/confluence_prepare_data.py [hunk 3]
```diff
@@ -21,17 +24,44 @@
 ENGLISH_GB = 'en_GB'
 
 
+def print_timing(message, sep='-'):
```
Comment (ometelytsia) [inline app/util/data_preparation/confluence_prepare_data.py:27]: This is a code duplication. jsm_prepare_data already has this. Need to extract this decorator to a common place, e.g. introduce new util file for prepare_data scripts.

File: app/util/data_preparation/confluence_prepare_data.py [hunk 4]
```diff
@@ -21,17 +24,44 @@
...
 def __create_data_set(rest_client, rpc_client):
     dataset = dict()
     dataset[USERS] = __get_users(rest_client, rpc_client, CONFLUENCE_SETTINGS.concurrency)
     perf_user = random.choice(dataset[USERS])['user']
     perf_user_api = ConfluenceRestClient(CONFLUENCE_SETTINGS.server_url, perf_user['username'], DEFAULT_USER_PASSWORD)
-    dataset[PAGES] = __get_pages(perf_user_api, 5000)
-    dataset[BLOGS] = __get_blogs(perf_user_api, 5000)
+
+    def __get_pages_blogs(func):
+        return func(perf_user_api, 5000)
```
Comment (ometelytsia) [inline app/util/data_preparation/confluence_prepare_data.py:59]: With this approach, we break the possibility to modify the number of pages and blogs separately.
Also, the code is quite hard to understand.

Let's try something more simple and straight forward like:
```
    pool = ThreadPool(processes=2)
    async_pages = pool.apply_async(__get_pages, (perf_user_api, 5000))
    async_blogs = pool.apply_async(__get_blogs, (perf_user_api, 5000))

    async_pages.wait()
    async_blogs.wait()

    dataset[PAGES] = async_pages.get()
    dataset[BLOGS] = async_blogs.get()
```

File: app/util/data_preparation/confluence_prepare_data.py [hunk 5]
```diff
@@ -25,13 +26,23 @@ def generate_random_string(length=20):
     return "".join([random.choice(string.ascii_lowercase) for _ in range(length)])
 
 
+@print_timing('Creating dataset started')
 def __create_data_set(rest_client, rpc_client):
-    dataset = dict()
+    dataset = dict([])
```
Comment (ometelytsia) [inline app/util/data_preparation/confluence_prepare_data.py:31]: Why this line was changed?

**Review by ometelytsia — APPROVED** [review]

File: app/util/data_preparation/confluence_prepare_data.py (hunk 5 above)
Comment (opopovss) [reply app/util/data_preparation/confluence_prepare_data.py:31]: probably it is just my mistake

**Review by mmizin — APPROVED** [review]

**Pull request merged**